src/fsw/FCCode/Drivers/Piksi.cpp

```diff
@@ -271,32 +307,34 @@ unsigned char Piksi::read_all() {
     #ifdef DESKTOP
     return _read_return;
     #else
+
+    noInterrupts();
+
+    int bytes = Serial4.available();
+    last_bytes = 0;

```

Comment:
Sorry I didn't notice this the first time but I think we should set `bytes` to the `last_bytes` seen by the interrupt. Otherwise, we could see data passed in between the interrupt and the start of this function call that doesn't get timestamped. It could also mess up timing for the next packet I think.
```suggestion
    int bytes = last_bytes;
    last_bytes = 0;
```